dali/pipeline/executor/executor2/exec2.h

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2023-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_EXECUTOR_EXECUTOR2_EXEC2_H_
+#define DALI_PIPELINE_EXECUTOR_EXECUTOR2_EXEC2_H_
+
+#include <memory>
+#include <string>
+#include <vector>
+#include "dali/pipeline/graph/op_graph2.h"
+#include "dali/pipeline/workspace/workspace.h"
+#include "dali/pipeline/executor/executor.h"
+
+namespace dali {
+namespace exec2 {
+
+enum class QueueDepthPolicy : int {
+  FullyBuffered,  //< All operators maintain a queue
+  BackendChange,  //< Only operators followed by one with a different backend have a queue
+  OutputOnly,     //< Only the pipeline output has multiple buffers
+  Legacy = BackendChange,
+};
```

Comment:
I found it hard to understand from this documentation (only this one, the other enums are clear). Could you extend the docs slightly and maybe back it up with some example?

Reply:
OK, I'll try to clarify.